internal/storage/storage_handle.go

```diff
@@ -61,6 +61,9 @@ func createGRPCClientHandle(ctx context.Context, clientConfig *storageutil.Stora
 
 	// Add Custom endpoint option.
 	if clientConfig.CustomEndpoint != nil {
+		if !clientConfig.SkipAuth {
```

Comment:
Skip auth is independent from TPC.

I think, we need to add line: 70 and 80 for gRPC if skipAuth is true.
Also, this should happen irrespective of custom-endpoint.

Reply:
> I think, we need to add line: 70 and 80 for gRPC if skipAuth is true.
Added conditions for both CustomEndpoint nil or non-nil
> this should happen irrespective of custom-endpoint.
For GCS endpoints, authentication will be skipped if 'Skip Auth' is enabled; otherwise, a tokenSrc will be added.